chore(release): add version 1.0.2 with new build method, Webpack setu…

sebastianjnuwu opened the pull request:
# Version 1.0.2 - New Improvements and Updates

In version 1.0.2, significant improvements have been made to the build process, code formatting, and linting configurations to ensure better organization and code quality. The changes include:

- **New Build Method with Webpack**: The build structure has been enhanced with **Webpack**, providing a more efficient way to bundle and optimize project resources.

- **Code Formatting with Biome**: The code now follows a more consistent formatting standard using **Biome**, ensuring a uniform appearance and better readability across the entire project.

- **Improved Linting Configuration**: A more robust linting configuration has been added to ensure code quality and quickly identify errors, style issues, or non-compliant patterns.

- **GitHub Action for Typos.toml**: A new **GitHub Actions** integration has been added, responsible for detecting and correcting spelling errors in the `.toml` configuration files, increasing the accuracy and efficiency of the project.

- **Documentation with JsDoc**: **JsDoc** comments have been added to the source code, providing clear and detailed explanations of functions, parameters, and return values, making the code easier to understand and maintain.

sebastianjnuwu committed:
chore(release): add version 1.0.2 with new build method, Webpack setup, Biome format, linting, and GitHub action

Added new version 1.0.2 with a new build method using Webpack, Biome code formatting, linting configuration, and a GitHub action for typos.toml. Also, added JsDoc comments in the code for better documentation.

sebastianjnuwu committed:
fix: deleting package-lock.json

To maintain code security and avoid installation problems on different devices and the aesthetics of the repository

sebastianjnuwu commented:
I recommend doing this to be found better:
![Screenshot_20250107_151123_Chrome](https://github.com/user-attachments/assets/ac7c7d66-1a6d-4cb0-8972-f5682d786dd7)

overskul commented on src/main.js:
```diff
@@ -166,32 +178,25 @@ class WakaTimePlugin {
-      await acodePlugin.init($page, cacheFile, cacheFileUrl);
-    },
-    acodePlugin.settingsObj
+    () => new WakaTimePlugin().init(),
```
you must declare a variable for the plugin

overskul commented on src/main.js:
```diff
@@ -166,32 +178,25 @@ class WakaTimePlugin {
-    },
-    acodePlugin.settingsObj
+    () => new WakaTimePlugin().init(),
+    new WakaTimePlugin().settingsObj,
```
creating a new class for setting obj not necessary, you can use variable instead

overskul commented on src/main.js:
```diff
@@ -166,32 +178,25 @@ class WakaTimePlugin {
-    acodePlugin.destroy();
-  });
+
+  acode.setPluginUnmount(plugin.id, () => new AcodeEruda().destroy());
```
this will throw error when uninstall the plugin and won't remove original plugin, please use variable

overskul commented on .acode/zip.js:
`package.json` missing `module` type.
and files path ig is wrong please check it

overskul commented on webpack.config.js:
`package.json` missing `module` type.

sebastianjnuwu committed:
chore(config): fix missing module type in package.json and incorrect file paths

- Added missing 'type' field in package.json to specify module type (module/commonjs).

- Corrected file paths in webpack.config.js and import statements to ensure proper resolution of files.

sebastianjnuwu commented:
```bash
npm run release 
```

```bash
npm run lint
```

```bash
npm run format 
```

```bash
typos --config typos.toml
```

sebastianjnuwu commented:
Any questions? I tested it here and it is working correctly!

overskul commented:
thanks for your great contribute
